conwayste/src/ui/common.rs

```diff
@@ -49,6 +45,35 @@ macro_rules! widget_from_id {
     }
 }
 
+/// A basic macro to downcast a `dyn Widget` to a concrete type. No turbo-fish required!
+///
+/// # Arguments
+/// Two required arguments are the `dyn Widget` object and the destination type.
+/// An optional mut argument, when included prior to the widget, returns a mutable reference.
+///
+/// # Usage
+/// ```Rust
+/// downcast_widget!([mut] widget, type) macro
+/// ```
+///
+/// # Examples
+/// ```rust
+/// let widget = layer.get_widget_mut(WidgetID(0));
+/// let textfield = downcast_widget!(mut widget, TextField);
+/// textfield.enter_focus()
+/// ```
+///
+#[macro_export]
+macro_rules! downcast_widget {
```

Comment:
Nice work -- the code looks cleaner with this macro! :grin: 

I think it looks a little strange to have `mut x` where you cannot do `x = <new value>` which can be the case with this macro.

It might be better as `downcast_widget!` and `downcast_widget_mut!`.